src/Microsoft.OData.Core/JsonLight/ODataJsonLightReader.cs

```diff
@@ -2061,6 +2066,13 @@ private void StartNestedResourceInfo(ODataJsonLightReaderNestedResourceInfo read
             ODataNestedResourceInfo nestedResourceInfo = readerNestedResourceInfo.NestedResourceInfo;
             IEdmProperty nestedProperty = readerNestedResourceInfo.NestedProperty;
             IEdmStructuredType targetResourceType = readerNestedResourceInfo.NestedResourceType;

```

Comment:
>NestedResourceType [](start = 77, length = 18)

maybe do the same change for the "NestedResourceType", I mean it should be "NestedResourceTypeReference", then you don't need to call "ToTypeReference(...)".